src/Context/FilterContext/FilterContext.js

```diff
@@ -0,0 +1,35 @@
+import { createContext, useContext, useReducer } from "react";
+import {
+  filterReducer,
+  initialFilterState,
+} from "../../Reducers/filterReducer";
+import { useNotes } from "../NotesContext/NotesContext";
+import {
+  Compose,
+  filterByLabel,
+  sortByPriority,
+} from "../../Utils/filterUtils";
+
+const FilterContext = createContext();
+const FilterProvider = ({ children }) => {
+  const { notesList } = useNotes();
+  const [filterState, dispatchFilter] = useReducer(
+    filterReducer,
+    initialFilterState
+  );
+
+  const finalFilteredNotes = Compose(
+    filterState,
+    sortByPriority,
+    filterByLabel
+  )(notesList);
+  return (
+    <FilterContext.Provider
+      value={{ filterState, dispatchFilter, finalFilteredNotes }}
```

Comment:
The name could be filteredNotes instead of finalFIlteredNoted, also filterDispatch instead of dispatchFilter.

No need to change here but try to keep naming consistent from here on